docs/adr/adr_gcloud-high-availability.md

```diff
@@ -0,0 +1,72 @@
+# ADR: GCloud high availability mode
+
+[Trello-card](https://trello.com/c/BYNLaDQh/1377-discuss-gcloud-high-availability-mode-incl-action)
+
+Decision Deadline: 2024-02-29
+
+Author: @pylipp
+
+## Status
+
+Proposed.
+
+## Context or Problem Statement
+
+We use GCloud to host the boxtribute MySQL database in the cloud. It can be accessed by the app (deployed to Google App Engine), and by developers via the Cloud SQL proxy. The server resides in the `europe-west1` region in the zone `europe-west1-d`.
+
+Our GCloud instance is configured in High Availability (HA) mode. Quoting the docs [1]:
+
+> The purpose of an HA configuration is to reduce downtime when a zone or instance becomes unavailable. This might happen during a zonal outage, or when there's a hardware issue. With HA, your data continues to be available to client applications.
+> The HA configuration provides data redundancy. A Cloud SQL instance configured for HA is also called a regional instance and has a primary and secondary zone within the configured region. Within a regional instance, the configuration is made up of a primary instance and a standby instance. Through synchronous replication to each zone's persistent disk, all writes made to the primary instance are replicated to disks in both zones before a transaction is reported as committed. In the event of an instance or zone failure, the standby instance becomes the new primary instance. Users are then rerouted to the new primary instance. This process is called a failover.
+
+For the statviz project, we introduced a *read replica* of the primary database instance to transfer the load for reading large data sets (relatively to the data read for common app usage). The read replica resides in the `europe-west1-c` zone. Furthermore [2]:
+
+> In a disaster recovery scenario, you can promote a replica to convert it to a primary instance. This way, you can use it in place of an instance that's in a region that's having an outage. You can also promote a replica to replace an instance that's corrupted.
+> The read replica is an exact copy of the primary instance. Data and other changes on the primary instance are updated in almost real time on the read replica.
+> Read replicas are read-only; you cannot write to them. The read replica processes queries, read requests, and analytics traffic, thus reducing the load on the primary instance.
+
+A primary instance cannot failover to a read replica, and read replicas are unable to failover in any way during an outage. [3]
+
+Our read replica does not have HA mode enabled itself.
+
+Finally, we have nightly back-ups configured on the primary database instance with point-in-time recovery enabled.
+
+## Decision Drivers
+
+1. **Data safety**: we want user and app data to persist even if a GCloud outage happens
```

Comment:
There are some GDPR requirements in here, too.

> 4.7 Availability and Resilience
The Processor has taken the following measures:
Nightly backup of the database.
Automated and manual testing before deployment to production instance of the Software.
Automatic notifications if an error occurs in the Software’s production instance.
The security measures of Subprocessor for hosting can be found here under section 7 and Appendix 2:
https://cloud.google.com/terms/data-processing-terms